packages/shared/src/components/PostOptionsMenu.tsx

```diff
@@ -64,18 +72,23 @@ export default function PostOptionsMenu({
     onRemovePost?.(_postIndex);
   };
 
-  const onReportPost = async (
+  const onReportPost: ReportPostAsync = async (
     reportPostIndex,
     reportedPost,
     reason,
     comment,
     blockSource,
   ): Promise<void> => {
-    reportPost({
+    const shouldProceed = await reportPost({
       id: reportedPost?.id,
       reason,
       comment,
     });
+
+    if (!shouldProceed) {
+      return;
+    }
```

Comment:
I'm not 100% sure if we should return and not fire the custom analytics?
What would cause the item to fail in this case?

Reply:
Was thinking about that as well but thought we require login (which is part of the `shouldProceed`) to avoid abuse, people can start just reporting and we wouldn't know from whom.